### PR TITLE
Bump Falcon to 1.0.0, make the test dependencies optional, add the Hug framework

### PR DESCRIPTION
**Pull request by berdario**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [X] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

framework

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -10778,35 +10778,45 @@ in modules // {
...
-    #});
-
     meta = {
       description = "An unladen web framework for building APIs and app backends";
       homepage = http://falconframework.org;
       license = licenses.asl20;
       maintainers = with maintainers; [ desiderius ];
     };
   });
+  hug = pkgs.python35Packages.buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: don't use `pkgs.python35Packages.`

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -10778,35 +10778,45 @@ in modules // {
...
+    name = "hug-2.1.2";
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/h/hug/${name}.tar.gz";
+      sha256 = "93325e13706594933a9afb0d4f0b0748134494299038f07df41152baf6f89f4c";
+    };
+
+    propagatedBuildInputs = with self; [ falcon requests2 ];
+
+    # tests are not shipped in the tarball
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: :+1:

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: cc @desiderius

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: don't use `pkgs.python35Packages.`

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: cc @desiderius

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: :+1:

**Comment by berdario** [comment]
I applied the change, but earlier we got this Travis error:

```
Collecting falcon==1.0.0 (from hug==2.1.2)

  Could not find a version that satisfies the requirement falcon==1.0.0 (from hug==2.1.2) (from versions: )

No matching distribution found for falcon==1.0.0 (from hug==2.1.2)

builder for ‘/nix/store/c68dns9avx9vkrai208npj2c8yxixyfz-python3.5-hug-2.1.2.drv’ failed with exit code 1

killing process 4271

error: build of ‘/nix/store/c68dns9avx9vkrai208npj2c8yxixyfz-python3.5-hug-2.1.2.drv’ failed

The invocation of "nix-build -A python35Packages.falcon -A python35Packages.hug -A python27Packages.hug -A python27Packages.falcon /home/travis/.nox/nixpkgs" failed

```

it seems a bit weird to me

I also tried to run `nox-review pr 18743` but I only get

```
[nix-shell:/tmp]$ nox-review pr 18743
=== Reviewing PR 18743 : Bump Falcon to 1.0.0, make the test dependencies optional, add the Hug framework
==> Fetching base (master)
==> Fetching PR
==> Fetching extra history for merging
==> Merging PR into base
error: Your local changes to the following files would be overwritten by checkout:
	pkgs/top-level/python-packages.nix
Please commit your changes or stash them before you switch branches.
Aborting
Traceback (most recent call last):
  File "/nix/store/7d0m6k9l94b58z0as0zppgz98bbain3l-nox-0.0.4/bin/.nox-review-wrapped", line 12, in <module>
    sys.exit(cli())
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/p50ngmmp4gfb3mdb8pcd8vjq0bzflggd-python3.5-click-6.6/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/nix/store/7d0m6k9l94b58z0as0zppgz98bbain3l-nox-0.0.4/lib/python3.5/site-packages/nox/review.py", line 135, in review_pr
    repo.checkout(base)
  File "/nix/store/7d0m6k9l94b58z0as0zppgz98bbain3l-nox-0.0.4/lib/python3.5/site-packages/nox/nixpkgs_repo.py", line 54, in checkout
    self.git(['checkout', '--quiet', sha])
  File "/nix/store/7d0m6k9l94b58z0as0zppgz98bbain3l-nox-0.0.4/lib/python3.5/site-packages/nox/nixpkgs_repo.py", line 48, in git
    return f(command, *args, cwd=cwd, universal_newlines=output, **kwargs)
  File "/nix/store/vvw059b8f6iz2yy6aj16x26af9f3clzy-python3-3.5.2/lib/python3.5/subprocess.py", line 581, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['git', 'checkout', '--quiet', 'cc22b5f2adc8eaf83b874847ef10e2bbd579f4b8']' returned non-zero exit status 1
```

**Comment by FRidh** [comment]
Oops. Here there were two packages in one commit. In the future, please separate them.

**Comment by berdario** [comment]
Sure

Thanks for the quick merge

**Pull request merged**